app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,44 @@
+class LikesController < ApplicationController
+  def up
+    if current_user.already_likes?(params[:post_id])
+      @like = current_user.likes.find(params[:like_id])
+      if @like.update(vote: true)
+        redirect_to posts_path
+      end
+    else          
+      @like = Like.new
```

Comment:
`@like = Like.new(user_id: current_user.id, post_id: params[:post_id], vote: true)` initializes `@like` with all the attributes.